jolt-core/src/jolt/vm/bytecode.rs

```diff
@@ -175,6 +216,44 @@ impl<F: PrimeField> FiveTuplePoly<F> {
             self.imm.evaluate(r),
         ]
     }
+
+    fn from_elf_r1cs(elf: &Vec<ELFRow>, N_SKIP: usize) -> Vec<F> {
+        // Do not pad. 
```

Comment:
Comment -> "Padding should be handled elsewhere."

Let's also assert the length is a power of 2 here.

Reply:
We don't need it to be a power of 2 for the R1CS polys. In fact, things get complicated when padding. 

We might need it when we feed polys directly into Spartan. but I'll just delete the comment for now.

Reply:
Can you provide more detail on the complication?

All of the polys that are derived from Lasso things are already powers of 2. So I think we should just assert that all of them are a power of 2 length as to avoid future bugs.

Reply:
Padding will make all the witnesses 0, which isn't a satisfying assignment for the R1CS. We can work around it and modify the R1CS constraints to "turn off" when we want it to but that'll take some effort. I think padding is also internally in Spartan so we'll have to revisit this when plugging polynomial commitments directly from Jolt to Spartan.